lib/core/fs/compiler/installed_compiler.dart

```diff
@@ -22,14 +29,14 @@ extension InstalledCompilerImpl on InstalledCompiler {
     return InstalledCompilerImpl._available[this]!;
   }
 
+  /// The path where the compiler is installed.
   Future<String?> path() async {
-    if (InstalledCompilerImpl._available.containsKey(this) &&
-        InstalledCompilerImpl._available[this] == false) {
-      return null;
+    if (InstalledCompilerImpl._available.containsKey(this)) {
```

Comment:
If available knows this, we already checked for availability. If available we have stored the location and can return it. Otherwise map will return null for an unknown key, which is also what path returns if not available.